### PR TITLE
Feature/java17image

### PR DESCRIPTION
**Pull request by arunaukumar**
# Description

This PR includes the upgradation of java version from 11 to 17 , we have used 17-jdk-jammy version .

# Checklist:

- [x] I have ensured the Defra version in the **JOB.env** file matches that in the **Dockerfile**
- [x] I have ensured the Java versions in the **image-matrix.json** match the **Dockerfile** and the table in the **README.md**
- [x] I have added newly ignored vulnerabilities to the **POLICY_CONFIGURATION.md**
- [x] I have checked if previously identified vulnerabilities have been patched, and can be removed from the **.grype.yaml** and **POLICY_CONFIGURATION.md** files

### CODE REVIEW

**Review by ben-sagar — CHANGES_REQUESTED** [review]
The PR needs to provide the purpose of the change, presumably it's something like

> Add support for a Java 17 image

It needs to (briefly) explain why 17 is being added.

It should also explain why a JRE base image is being used as previously JDK images were used.

**Comment by DaRoszko** [comment]
> The PR needs to provide the purpose of the change, presumably it's something like
> 
> > Add support for a Java 17 image
> 
> It needs to (briefly) explain why 17 is being added.
> 
> It should also explain why a JRE base image is being used as previously JDK images were used.

@arunaukumar  can you confirm that

File: Dockerfile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Set default values for build arguments
-ARG DEFRA_VERSION=1.1.1
-ARG BASE_VERSION=11-jdk-jammy
+ARG DEFRA_VERSION=1.1.2
+ARG BASE_VERSION=17-jre-jammy
```
Comment (DaRoszko) [inline Dockerfile:3]: should this be sdk instead?

**Review by DaRoszko — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (arunaukumar) [reply Dockerfile:3]: choose JRE instead of JDK - no development tools on the image, just the ones needed for execution

Comment (PriceChild) [reply Dockerfile:3]: @arunaukumar Be careful, I'm reasonably certain that at least one of your consumers requires the JDK for their execution images.

That said, this should be a safe (if unnecessary change) as all images will be produced still due to the image-matrix.json shenanigans which overrides the argument. The value in the Dockerfile is just a default for local development of the repository.

**Comment by arunaukumar** [comment]
Fixed it , to jdk 17


Warm Regards,
***@***.***Arun Kumar
Capgemini | India | Gurugram
Mob.: + 91 9888 079 170
Upcoming Holiday:-
Upcoming Leave:- 8th to 11th Dec’23
***@***.******@***.******@***.******@***.******@***.***

**Comment by arunaukumar** [comment]
moved to jdk-17

**Comment by arunaukumar** [comment]
> The PR needs to provide the purpose of the change, presumably it's something like
> 
> > Add support for a Java 17 image
> 
> It needs to (briefly) explain why 17 is being added.
> 
> It should also explain why a JRE base image is being used as previously JDK images were used.

Resolved

**Pull request closed without merging**